modules/tensor_mechanics/src/materials/HillPlasticityStressUpdate.C

```diff
@@ -125,71 +139,78 @@ ADHillPlasticityStressUpdate::computeDeltaDerivatives(const ADReal & delta_gamma
   sy_gamma = 2.0 * sy_alpha * (omega + delta_gamma * omega_gamma);
 }
 
+template <bool is_ad>
 Real
-ADHillPlasticityStressUpdate::computeReferenceResidual(
-    const ADDenseVector & /*effective_trial_stress*/,
-    const ADDenseVector & /*stress_new*/,
-    const ADReal & /*residual*/,
-    const ADReal & /*scalar_effective_inelastic_strain*/)
+HillPlasticityStressUpdateTempl<is_ad>::computeReferenceResidual(
+    const GenericDenseVector<is_ad> & /*effective_trial_stress*/,
+    const GenericDenseVector<is_ad> & /*stress_new*/,
+    const GenericReal<is_ad> & /*residual*/,
+    const GenericReal<is_ad> & /*scalar_effective_inelastic_strain*/)
 {
   // Equation is already normalized.
   return 1.0;
 }
 
-ADReal
-ADHillPlasticityStressUpdate::computeResidual(const ADDenseVector & stress_dev,
-                                              const ADDenseVector & /*stress_sigma*/,
-                                              const ADReal & delta_gamma)
+template <bool is_ad>
+GenericReal<is_ad>
+HillPlasticityStressUpdateTempl<is_ad>::computeResidual(
+    const GenericDenseVector<is_ad> & stress_dev,
+    const GenericDenseVector<is_ad> & /*stress_sigma*/,
+    const GenericReal<is_ad> & delta_gamma)
 {
 
   // If in elastic regime, just return
   if (_yield_condition <= 0.0)
     return 0.0;
 
-  ADDenseMatrix eigenvectors_hill_transpose(6, 6);
+  GenericDenseMatrix<is_ad> eigenvectors_hill_transpose(6, 6);
 
   _eigenvectors_hill.get_transpose(eigenvectors_hill_transpose);
   eigenvectors_hill_transpose.vector_mult(_stress_np1, stress_dev);
 
-  ADReal omega = computeOmega(delta_gamma, _stress_np1);
+  GenericReal<is_ad> omega = computeOmega(delta_gamma, _stress_np1);
   _hardening_slope = computeHardeningDerivative();
 
   // Hardening variable is \alpha isotropic hardening for now.
   _hardening_variable[_qp] = computeHardeningValue(delta_gamma, omega);
-  ADReal s_y = _hardening_slope * _hardening_variable[_qp] + _yield_stress;
+  GenericReal<is_ad> s_y = _hardening_slope * _hardening_variable[_qp] + _yield_stress;
 
-  ADReal residual = 0.0;
+  GenericReal<is_ad> residual = 0.0;
   residual = s_y / omega - 1.0;
 
   return residual;
 }
 
-ADReal
-ADHillPlasticityStressUpdate::computeDerivative(const ADDenseVector & /*stress_dev*/,
-                                                const ADDenseVector & /*stress_sigma*/,
-                                                const ADReal & delta_gamma)
+template <bool is_ad>
+GenericReal<is_ad>
+HillPlasticityStressUpdateTempl<is_ad>::computeDerivative(
```

Comment:
I am puzzled as to why this method, for this object (i.e. `HillPlasticityStressUpdate`) is never hit in coverage: Don't the existing tests exercise the mapping? Any idea?

Reply:
That would occur when there is no plasticity occurring. Not sure, will need to look into it.